### ISSUE TITLE
Variables should reflect changes made by the user in the debug console

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/PowerShellEditorServices#58]
If the user modifies, add or removes a variable in the debug console during a debug stop, that should be reflected after execution of their command.  

We will probably need to re-fetch the variables as we can't rely on examining the executed expression for simple variable assignment. The user could be using Add/Clear/Remove/Set-Variable or they may be manipulating the variable drive.

**Comment by daviwil** [comment]
One problem we'll have here is that VS Code doesn't give us a way to notify the editor that variables have changed need to be refreshed.  We may be able to ask for this though!

**Comment by rkeithhill** [comment]
OK that has been addressed by the VSCode team.  Yay!
![vscodechangedvariable](https://cloud.githubusercontent.com/assets/5177512/12766469/3ae1fbc0-c9c1-11e5-855e-b3d64d6e6f14.png)

**Comment by daviwil** [comment]
Awesome news.

**Comment by rkeithhill** [comment]
Oops re-opening this as it is a separate issue from the variable dirty indicator and it is not fixed.  :-)

**Comment by daviwil** [comment]
Ahhh I guess it's not being updated when the user types in something that changes a variable value :)  Makes sense.  We'd need some way to signal variable changes to them.  I wonder if this is even something they'd be interested in adding?  Will probably require a feature request here:

https://github.com/Microsoft/vscode-debugadapter-node

**Comment by rkeithhill** [comment]
If they want to better support REPL friendly languages I think they would want this. I also think it would be slick if they allowed the user to edit a variable's value directly in the "variables" tree view while the debugger is stopped.  Of course, the debug adapter would need to support such an operation.

**Comment by daviwil** [comment]
I seem to recall that they have the desire to allow variable editing, but I can't remember if that was a bug on the VS Code repo or from some e-mail thread.  Either way, I think you're right, they'll definitely need to have the variable tree up to date for REPL friendly languages.

**Comment by tylerl0706** [comment]
Circling back to this. If I modify the value in the debug console and step over in the debugger, the value does change:

![image](https://user-images.githubusercontent.com/2644648/33736194-8afd1a52-db46-11e7-8455-a7ada7f51e73.png)

Is this what this issue was about or was it more about changing the variable and have it immediately updated in the debug pane?

**Comment by daviwil** [comment]
Right, they don't give us a way to notify the UI that the variable has changed "out of band" so there's currently not a way to accomplish this (that I'm aware of).

**Comment by andyleejordan** [comment]
Resolved — variable edits in the debug console are applied via `SetVariableAsync`/`SetVariableHandler` (covered by `DebugServiceTests`). Closing as completed.

---
_Drafted by GitHub Copilot (Claude Opus 4.8) at @andyleejordan's request._
